tuichain/api/views/loans.py

```diff
@@ -361,8 +364,30 @@ def get_loan(request, id):
 
         loan_state = fetched_loan.get_state()
 
+        if loan_state.phase in [
+            LoanPhase.FUNDING,
+            LoanPhase.CANCELED,
+            LoanPhase.EXPIRED,
+        ]:
+            value_atto_dai = 10 ** 18
+
+        elif loan_state.phase == LoanPhase.ACTIVE:
+            try:
+                value_atto_dai = median(
+                    repeat(sp.price_atto_dai_per_token, sp.amount_tokens)
+                    for sp in controller.market.get_sell_positions_by_loan(
+                        fetched_loan
+                    )
+                )
+            except statistics.StatisticsError:
+                value_atto_dai = None
+
+        else:  # loan_state.phase == LoanPhase.FINALIZED
+            value_atto_dai = loan_state.redemption_value_atto_dai_per_token
+
         loan_dict["state"] = loan_state.phase.name
         loan_dict["funded_value_atto_dai"] = loan_state.funded_value_atto_dai
+        loan_dict["current_price_atto_dai"] = value_atto_dai
```

Comment:
Dai values must be serialized as strings to avoid precision loss. Also remember to handle the `value_atto_dai = None` case correctly.